lib/rules/selector-class-pattern/index.js

```diff
@@ -110,16 +110,20 @@ const rule = (primary, secondaryOptions) => {
  * @returns {boolean}
  */
 function hasInterpolatingAmpersand(selector) {
-	for (let i = 0, l = selector.length; i < l; i++) {
-		if (selector[i] !== '&') {
+	for (const [i, char] of Array.from(selector).entries()) {
```

Comment:
[note] [`Array.from()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from) can make a string to an array of characters.